### PR TITLE
ovl_En_Horse_Link_Child

### PR DESCRIPTION
**Pull request by maekclena**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 1]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+
+    if (gSaveContext.entranceIndex == 0x2AE) {
+        Audio_PlaySoundGeneral(NA_SE_EV_KID_HORSE_NEIGH, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        func_80A6A724(this);
+        return;
+    }
+
+    if (((gSaveContext.eventChkInf[1] & 0x40) && (DREG(53) != 0)) ||
+        ((globalCtx->sceneNum == SCENE_SPOT20) && (gSaveContext.cutsceneIndex == 0xFFF1))) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:329]: The & 0x40 can be & (1<<6)

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 2]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+        func_80A6A4DC(this);
+    } else {
+        this->unk_2A0 = gSaveContext.eventChkInf[1] & 0x40;
+    }
+
+    newAnimationIdx = this->animationIdx;
+    animationEnded = SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime);
+    if (animationEnded || (this->animationIdx == 1) || (this->animationIdx == 0)) {
+        if (gSaveContext.eventChkInf[1] & 0x20) {
+            distFromHome = Math3D_Vec3f_DistXYZ(&this->actor.posRot.pos, &this->actor.initPosRot.pos);
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:339]: & 0x20 can be & (1<<5)

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 3]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+    Player* player = PLAYER;
+    f32 dist;
+    s32 newAnimationIdx;
+
+    this->timer++;
+    if (this->timer > 300) {
+        this->unk_1E8 = true;
+    }
+
+    if ((this->animationIdx == 4) || (this->animationIdx == 3) || (this->animationIdx == 2)) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:458]: The animation indicesay have an enum

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 4]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A6988C.s")
+static EnHorseLinkChildActionFunc D_80A6AF7C[] = { func_80A698F4, func_80A69C18, func_80A699FC,
+                                                   func_80A6A068, func_80A6A7D0, func_80A6A5A4 };
+static UNK_PTR D_80A6AF94[] = { 0x06001D28, 0x06001928, 0x06001B28 };
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:515]: What type are these?

**Review by Kenix3 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c (hunk 1 above)
Comment (fig02) [reply src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:329]: as noted in other prs, hex is fine

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c (hunk 4 above)
Comment (fig02) [reply src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:515]: they are probably textures, but its not info you can get from the code alone which is why unk_ptr was used

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c (hunk 3 above)
Comment (fig02) [reply src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:458]: an enum can be made if its known what they correspond to, but this is a documentation thing that isnt required right now

**Review by Kenix3 — APPROVED** [review]

**Review by NEstelami — APPROVED** [review]

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 5]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+static CollisionCheckInfoInit sColCheckInfoInit = { 10, 35, 100, 0xFE };
+
+extern AnimationHeader D_06002F98;
+extern SkeletonHeader D_06007B20;
+
+void func_80A693D0(EnHorseLinkChild* this) {
+    static s32 D_80A6AF5C[] = { 1, 19 };
+
+    if ((D_80A6AF5C[this->unk_1F0] < this->skin.skelAnime.animCurrentFrame) &&
+        !((this->unk_1F0 == 0) && (D_80A6AF5C[1] < this->skin.skelAnime.animCurrentFrame))) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:57]: if it matches, I would put the array accesses on the other side of the comparison
`if (frame count is > array accesses)`

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 6]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+            func_8006DD9C(&this->actor, &player->actor.posRot.pos, 300);
+        } else {
+            func_8006DD9C(&this->actor, &this->actor.initPosRot.pos, 300);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A695A4.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        if (!this->unk_1E8) {
+            dist = func_8002DB8C(&this->actor, &PLAYER->actor);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:468]: i assume PLAYER directly was necessary here

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c [hunk 7]
```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
...
+        } else {
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A6988C.s")
+static EnHorseLinkChildActionFunc D_80A6AF7C[] = { func_80A698F4, func_80A69C18, func_80A699FC,
+                                                   func_80A6A068, func_80A6A7D0, func_80A6A5A4 };
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:514]: you can put a comma on the last entry to stop the indentation. and you can call it `sActionFuncs` if you want

File: src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c (hunk 6 above)
Comment (maekclena) [reply src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c:468]: It is. `PLAYER` is accessed twice and I had no luck with other combinations.

**Review by fig02 — APPROVED** [review]

**Pull request merged**